sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2616,4 +2617,66 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     val e = intercept[AnalysisException](sql("SELECT nvl(1, 2, 3)"))
     assert(e.message.contains("Invalid number of arguments"))
   }
+
+  test("TRIM function-BOTH") {
+    val ae1 = intercept[ParseException]{
+      sql("select lower(BOTH 'S' FROM 'SS abc S')").head
```

Comment:
The keyword BOTH/LEADING/TRAILING/FROM can only be used by trim function. I was thinking to show there will be throwing exceptions if other functions try to use these keywords.  I can change lower --> LTRIM/RTRIM, what do you think?